btx/interfaces/ipsana.py

```diff
@@ -109,12 +112,11 @@ def get_wavelength_evt(self, evt):
         wavelength : float
             wavelength in Angstrom
         """
-        photon_energy = self.get_photon_energy_eV_evt(evt)
-        if np.isinf(photon_energy) or photon_energy is None:
-            return self.get_wavelength()
-        else:
-            lambda_m =  1.23984197386209e-06 / photon_energy # convert to meters using e=hc/lambda
+        try:
+            lambda_m = 1.23984197386209e-06 / self.get_photon_energy_eV_evt(evt) # convert to meters using e=hc/lambda
             return lambda_m * 1e10
+        except:
```

Comment:
If we go with the try/except, we should narrow it to the specific error we're anticipating, i.e. `TypeError`

Reply:
Let's use conditional if that works